far/vmenu.cpp

```diff
@@ -2011,79 +2097,121 @@ int VMenu::VisualPosToReal(int VPos) const
 	return ItemIterator != Items.cend()? ItemIterator - Items.cbegin() : -1;
 }
 
-size_t VMenu::GetItemMaxShowPos(int Item, const size_t MaxLineWidth) const
+template<typename GetNewHPos_T>
+bool VMenu::SetItemHPos(MenuItemEx& Item, const GetNewHPos_T GetNewHPos, int const TextAreaWidth)
```

Comment:
- Is `TextAreaWidth` used anywhere?
- This syntax with `template<>` in cpp, but `auto` in hpp is a little mind-blowing: I spent like 15 sec trying to understand what I'm even looking at.
Not sure yet do I love it or hate it.
It makes perfect sense though, since GetNewHPos_T is not needed in hpp and needed here, so... I guess I like it.

Reply:
- Ugh! Good catch! `TextAreaWidth` is a leftover from previous experiments. Removed it and it nicely rippled out.
- Yes, it was to avoid verbose syntax in the header, but the `requires` syntax here wins hands down.